composio/utils/shared.py

```diff
@@ -93,11 +90,32 @@ def json_schema_to_pydantic_field(
         Field(
             description=description,
             examples=examples,
-            default=... if name in required else None,
+            default=... if name in required else default,
         ),
     )
 
 
+def json_schema_to_fields_dict(json_schema: t.Dict[str, t.Any]) -> t.Type[BaseModel]:
+    """
+    Converts a JSON schema to a dictionary of param name, and a tuple of type & Field.
+
+    :param json_schema: The JSON schema to convert.
+    :return: dict<str, tuple<<class 'type'>, Field>>
+
+    Example Output:
+    {
+        'owner': (<class 'str'>, FieldInfo(default=Ellipsis, description='The account owner of the repository.', extra={'examples': ([],)})), 
+        'repo': (<class 'str'>, FieldInfo(default=Ellipsis, description='The name of the repository without the `.git` extension.', extra={'examples': ([],)}))}
+    }
+
+    """
```

Comment:
```suggestion
    Example:
    ```python
    >>> json_schema_to_fields_dict()
    {
        'owner': (<class 'str'>, FieldInfo(default=Ellipsis, description='The account owner of the repository.', extra={'examples': ([],)})), 
        'repo': (<class 'str'>, FieldInfo(default=Ellipsis, description='The name of the repository without the `.git` extension.', extra={'examples': ([],)}))}
    }
    ```
    """
```

(This is just a suggested format for writing docstring, you can ignore if you want to)

Reply:
taken into consideration and modified.